pytest_html/plugin.py

```diff
@@ -64,10 +69,11 @@ def pytest_configure(config):
     htmlpath = config.option.htmlpath
     # prevent opening htmlpath on slave nodes (xdist)
     if htmlpath and not hasattr(config, 'slaveinput'):
-        config._html = HTMLReport(htmlpath,
-                                  config.getoption('self_contained_html'),
-                                  config.pluginmanager
-                                  .hasplugin('rerunfailures'))
+        config._html = HTMLReport(
+            htmlpath,
+            config.getoption('self_contained_html'),
+            config.pluginmanager.hasplugin('rerunfailures'),
+            config)
```

Comment:
Since you pass the `config` object, you can remove `self_contained` and `has_rerun` (the latter in `TestResult` too).